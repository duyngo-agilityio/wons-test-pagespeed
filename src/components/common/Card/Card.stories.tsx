import type { Meta, StoryObj } from '@storybook/react';

// components
import Card from './index';

const meta: Meta<typeof Card> = {
  title: 'Components/Common/Card',
  component: Card,
  argTypes: {
    rounded: {
      control: 'select',
      options: ['0px', '10px', '20px', '30px', '50%'],
    },
    bgColor: {
      control: 'select',
      options: ['pink', 'lightblue', 'lightgreen', '#ff6347', '#000'],
    },
    padding: {
      control: 'select',
      options: ['5px', '10px', '15px', '20px', '25px'],
    },
    borderColor: {
      control: 'select',
      options: ['transparent', 'black', 'red', 'blue', 'green'],
    },
    hoverEffect: {
      control: 'boolean',
    },
    header: {
      control: 'text',
      description: 'Content for the card header',
    },
    body: {
      control: 'text',
      description: 'Content for the card body',
    },
    footer: {
      control: 'text',
      description: 'Content for the card footer',
    },
    image: {
      control: 'text',
      description: 'URL for the image displayed in the card body',
    },
  },
};

export default meta;
type Story = StoryObj<typeof Card>;

export const Default: Story = {
  args: {
    bgColor: 'pink',
    rounded: '20px',
    padding: '10px',
    borderColor: 'transparent',
    hoverEffect: true,
    header: 'Card Header: This is custom card header',
    body: 'Card Body: This is custom card body',
    footer: 'Card Footer: This is custom card footer',
    image:
      'https://img-cdn.pixlr.com/image-generator/history/65bb506dcb310754719cf81f/ede935de-1138-4f66-8ed7-44bd16efc709/medium.webp',
  },
};
