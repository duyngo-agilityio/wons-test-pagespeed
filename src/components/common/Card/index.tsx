'use client';
import { memo } from 'react';
import {
  Card as CardNextUI,
  CardProps,
  CardHeader,
  CardBody,
  CardFooter,
} from '@nextui-org/react';

// components
import { ImageFallback } from '@/components';

interface CustomCardProps extends CardProps {
  rounded?: string;
  bgColor?: string;
  padding?: string;
  borderColor?: string;
  hoverEffect?: boolean;
  header?: React.ReactNode;
  body?: React.ReactNode;
  footer?: React.ReactNode;
  image?: string;
}

const Card = ({
  rounded,
  bgColor,
  padding,
  borderColor,
  hoverEffect,
  header,
  body,
  footer,
  image,
  ...props
}: CustomCardProps) => (
  <CardNextUI
    {...props}
    style={{
      borderRadius: `${rounded}`,
      backgroundColor: `${bgColor}`,
      padding: `${padding}`,
      border: `1px solid ${borderColor}`,
      boxShadow: hoverEffect ? `0 4px 6px rgba(0, 0, 0, 0.1)` : 'none',
    }}
  >
    {header && <CardHeader>{header}</CardHeader>}
    <CardBody>
      {body}
      {image && (
        <ImageFallback src={image} alt="Card image" width={100} height={100} />
      )}
    </CardBody>
    {footer && <CardFooter>{footer}</CardFooter>}
  </CardNextUI>
);

export default memo(Card);
