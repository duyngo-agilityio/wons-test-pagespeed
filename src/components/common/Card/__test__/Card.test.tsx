// components
import Card from '../index';

describe('Card component', () => {
  it('renders correctly and matches snapshot', () => {
    const { container } = testLibJestUtils.render(<Card />);
    expect(container).toMatchSnapshot();
  });
});
