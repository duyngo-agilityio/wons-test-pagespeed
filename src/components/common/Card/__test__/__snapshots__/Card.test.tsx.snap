// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly and matches snapshot 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
    <div
      class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none"
      style="border-radius: undefined; box-shadow: none;"
      tabindex="-1"
    >
      <div
        class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased"
      />
    </div>
  </div>
</div>
`;
